### PR TITLE
oauth2server: test fix

### PR DESCRIPTION
**Pull request by crepererum**
* Fixes oauth2server test.

Signed-off-by: Marco Neumann <marco@crepererum.net>

### CODE REVIEW

File: invenio/modules/oauth2server/testsuite/test_provider.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def test_invalid_authorize_requests(self):
             ))
             self.assertStatus(r, 302)
             next_url, data = self.parse_redirect(r.location)
-            self.assertEqual(data['error'], 'unauthorized_client')
```
Comment (jirikuncar) [inline invenio/modules/oauth2server/testsuite/test_provider.py]: IMHO you should not change the test. (cc @lnielsen)

**Comment by crepererum** [comment]
OK, so @hachreak tracked down the issue: [oauthlib](https://github.com/idan/oauthlib) released a new version which is not compatible with our tests. Because this lib is a second order requirement through [Flask-OAuthlib](https://github.com/lepture/flask-oauthlib), our `minimal` version generator does not work and all 3 test levels are failing. We now have the following options:
 1. skip the tests and fix it later (again: the tests seem to be broken, not the lib)
 2. add `oauthlib` to setup.py and put a constraint on it.

This is also related to #2158 which suggest that we should not constraint the version but rather get our tests fixed and finally do the upgrade for `Flask-OAuthlib` as well.

**Comment by jirikuncar** [comment]
> 2. add oauthlib to setup.py and put a constraint on it.

I would go for number two. WDYT @lnielsen

**Comment by lnielsen** [comment]
I vote for option 2 as well, unless someone wants to upgrade Flask-OAuthlib. Disabling the test is not an option in my opinion.

Btw, "Fixes oauth2server test." is quite misleading when in fact it's disabling the test ;-)

**Pull request closed without merging**